### PR TITLE
Feature/example project

### PR DESCRIPTION
**Pull request by ektaghag-eaton**
Fixes 4428 .

#### Changes proposed in this Pull Request:

- Created a new project with Common AuthContextProvider and RegistrationContextProvider
- Added homepage after login
-

#### Screenshots / Screen Recording (if applicable)

- 
Screenshot 2023-07-12 at 7 24 15 PM

Screenshot 2023-07-12 at 7 23 18 PM

#### To Test:

- run yarn start:example3 inside login-workflow folder

#### Any specific feedback you are looking for?

- Folder structure
- Routes arrangement
- Guarded routes
- **Fixing of Refreshing guarded page redirection is in progress (for e.g after login if I am on homePage and refresh the page it redirects to login ideally it should be on same page)**

### CODE REVIEW

**Comment by ektaghag-eaton** [comment]
Right now, we are not able to maintain the changed language and user's authorization in appContextProvider. Because of this if we refresh the homepage it redirects to login instead of being on the same page. Need a different story to fix this.

**Review by joebochill — COMMENTED** [review]
- Rename `ExampleProvider` to `AppRouter`
- Set up I18n (user-side) in the example project
- Make language picker part of the debug elements
- For UserMenu in home page, use menuGroups to preserve our default styling of the menu instead of passing in a completely custom Menu

**Comment by ektaghag-eaton** [comment]
> * Rename `ExampleProvider` to `AppRouter` - **Done**
> * Set up I18n (user-side) in the example project - **Done**
> * Make language picker part of the debug elements - **Done**
> * For UserMenu in home page, use menuGroups to preserve our default styling of the menu instead of passing in a completely custom Menu - **Done**

@joebochill please review

File: login-workflow/example3/src/App.tsx [hunk 1]
```diff
@@ -1,12 +1,13 @@
...
 import { BrowserRouter } from 'react-router-dom';
-import { ExampleProvider } from './screens';
+import { AppRouter } from './screens';
 import { SecurityContextProvider } from '@brightlayer-ui/react-auth-workflow';
+import './translations/i18n';
 
 export const App = (): JSX.Element => {
     const [isAuthenticated, setIsAuthenticated] = useState(false);
-    const [language, setLanguage] = useState('en');
+    const [language, setLanguage] = useState('es');
```
Comment (joebochill) [inline login-workflow/example3/src/App.tsx:10]: leave the default as English

File: login-workflow/example3/src/components/DebugComponent.tsx [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+            <Box
+                sx={{
+                    display: 'flex',
+                    flexDirection: 'row',
+                    justifyContent: 'space-between',
+                    alignItems: 'center',
+                }}
+            >
+                <Box>{debugMode && <Typography variant={'h6'}>DEBUG MODE</Typography>}</Box>
+                <Button
```
Comment (joebochill) [inline login-workflow/example3/src/components/DebugComponent.tsx:34]: Make this button position: absolute so it doesn't cause the login screen to scroll when the debug controls are hidden

File: login-workflow/example3/src/screens/ExampleHome.tsx [hunk 3]
```diff
@@ -89,20 +87,26 @@ export const ExampleHome: React.FC<React.PropsWithChildren> = () => {
...
-                                    >
-                                        {`${t('Blui:LOGOUT')}`}
-                                    </MenuItem>
-                                </Menu>
-                            }
+                            avatar={<Avatar>AV</Avatar>}
+                            menuGroups={[
+                                {
+                                    items: [
+                                        {
```
Comment (joebochill) [inline login-workflow/example3/src/screens/ExampleHome.tsx:94]: Add the icons to these menuItems

Comment (joebochill) [reply login-workflow/example3/src/screens/ExampleHome.tsx:94]: Consider adding a language dropdown menu too next to the user menu (not necessary, but if it's quick and easy, go for it)

File: login-workflow/package.json [hunk 4]
```diff
@@ -8,8 +8,7 @@
     "scripts": {
         "install:dependencies": "cd example && yarn && cd .. && yarn",
         "install:dependencies2": "cd example2 && yarn && cd .. && yarn",
-        "install:dependencies3": "cd example3 && yarn && cd .. && yarn",
-        "test": "jest src",
+        "test": "jest",
```
Comment (joebochill) [inline login-workflow/package.json]: add back the install 3 command...yarn start:example3 is broken now

**Review by joebochill — COMMENTED** [review]
There's some errors that need to be cleaned up:
image

**Review by surajeaton — APPROVED** [review]
Looks good.

**Pull request merged**